src/Language/PureScript/AST/Binders.hs

```diff
@@ -53,6 +53,16 @@ data Binder
   --
   | NamedBinder Ident Binder
   -- |
+  -- Binary operator application. During the rebracketing phase of desugaring, this data constructor
+  -- will be removed.
+  --
+  | BinaryNoParensBinder (Qualified Ident) Binder Binder
+  -- |
+  -- Explicit parentheses. During the rebracketing phase of desugaring, this data constructor
+  -- will be removed.
+  --
+  | ParensInBinder Binder
+  -- |
```

Comment:
I haven't really made use of this yet, as the parsing is not correct, but I assume we'll need something like this to enable fixity overriding as with the normal operators?